crypto/sleep.c

```diff
@@ -12,30 +12,53 @@
 
 /* system-specific variants defining OSSL_sleep() */
 #if defined(OPENSSL_SYS_UNIX) || defined(__DJGPP__)
-#include <unistd.h>
 
+# if defined(OPENSSL_USE_USLEEP)                        \
+    || defined(__DJGPP__)                               \
+    || (defined(__TANDEM) && defined(_REENTRANT))
+
+/*
+ * usleep() was made obsolete by POSIX.1-2008, and nanosleep()
+ * should be used instead.  However, nanosleep() isn't implemented
+ * on the platforms given above, so we still use it for those.
+ * Also, OPENSSL_USE_USLEEP can be defined to enable the use of
+ * usleep, if it turns out that nanosleep() is unavailable.
+ */
+
+#  include <unistd.h>
 void OSSL_sleep(uint64_t millis)
 {
-# ifdef OPENSSL_SYS_VXWORKS
-    struct timespec ts;
+    unsigned int s = (unsigned int)(millis / 1000);
+    unsigned int us = (unsigned int)((millis % 1000) * 1000);
+
+    if (s > 0)
+        sleep(s);
+    usleep(us);
```

Comment:
Should there be a check for `if (us > 0)` ?

Reply:
```suggestion
    if (s > 0)
        sleep(s);
    if (us > 0)
        usleep(us);
```

Reply:
This was discussed in multiple prior issues. There must be at least one thread context switch. If the `usleep()` is suppressed, alternate threads in PUT (thread model) will not be scheduled for execution. Please do not do this.

Reply:
i.e. @levitte would you mind adding a comment here that the `usleep(0)` is required so that there is at least one thread context switch?

Reply:
Sure, will do when I'm by my laptop

Reply:
[ahem] I forgot about this PR for a time.  Comment added now.